db/log_writer.cc

```diff
@@ -62,7 +63,16 @@ IOStatus Writer::Close() {
 }
 
 IOStatus Writer::AddRecord(const Slice& slice,
-                           Env::IOPriority rate_limiter_priority) {
+                           Env::IOPriority rate_limiter_priority,
+                           const std::map<uint32_t, size_t>* cf_to_ts_sz) {
+  IOStatus s;
+  if (cf_to_ts_sz != nullptr && !cf_to_ts_sz->empty()) {
+    s = MaybeAddUserDefinedTimestampSizeRecord(*cf_to_ts_sz,
+                                               rate_limiter_priority);
+    if (!s.ok()) {
+      return s;
+    }
+  }
```

Comment:
That's for the suggestion. I originally thought these timestamp size record should always be read and interpreted if it's found since it's part of correctly interpreting the WriteBatch, so I make it part of the ReadRecord API, and part of AddRecord API in mirror. This is not as clean because these parameters don't apply to manifest log. 

While working on the follow up change, I found some legitimate read record use cases that don't need the timestamp size record even if it's available. Like WalManager only read the sequence number for the first WriteBatch to get sorted WAL logs etc. So I agree more that separating the timestamp size record API makes sense.